### DISCUSSION TITLE
Parse additional fields on file uploads?

### DISCUSSION THREAD

**Discussion by BobbyWibowo** [kartikk221/hyper-express#89] [Q&A] [answered]
Is it currently not possible to parse additional fields from a file upload multiform via `res.multiform()` aside from the field containing the files themselves?

For example, suppose you use [Dropzone.js](https://github.com/dropzone/dropzone) and use its chunked uploads feature, an individual file will then be split into multiple chunks and gets uploaded one by one in their own multiform POSTs
Each chunk's multiform will look something like this (raw):
```
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="dzuuid"

f01e0370-57c0-435e-9e90-d6aa384b2662
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="dzchunkindex"

0
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="dztotalfilesize"

2609724
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="dzchunksize"

1000000
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="dztotalchunkcount"

3
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="dzchunkbyteoffset"

0
-----------------------------12601562561472663107612597899
Content-Disposition: form-data; name="files[]"; filename="Houkai.3rd.full.3397836.png"
Content-Type: application/octet-stream

PNG

...truncated
```

Back with ExpressJS + Multer, those additional fields could be read from `Response.body`, thus giving me enough data to implement custom "chunks re-combiner" in the backend (i.e. piping all the streams into a single write stream, in sequence, keyed by the uuid field sent by Dropzone.js)

With Hyper-Express' `res.multiform()`, I seemingly only have access to `files[]` field and the files themselves

**Comment by kartikk221** [comment, marked as answer]
HyperExpress currently does not support Chunked Encoding Transfer requests meaning the content-length of the body must be specified.

You can retrieve both normal text and file type fields with the Response.multipart() method as listed in the documentation.

HyperExpress uses the BusBoy multipart form parser under the hood which will call the handler with both normal and file containing fields.

**Reply by BobbyWibowo** [reply]
AH! I SEE NOW! The `field` callback is called per individual field!

For some reason I had the impression it would only be called once, ever, lmfao
My bad, I was logging too much unrelated data into the console that I missed them

But yeah, it's nothing as complicated as chunked encoding transfer, so no problem with that
In fact after figuring out how `res.multipart()` actually works, I already successfully ported the relevant codes and it works as expected

Yet once again, many thanks for this awesome project
